### DISCUSSION TITLE
Will o1 models be supported?

### DISCUSSION THREAD

**Discussion by REDDKOMET** [AllYourBot/hostedgpt#528] [General]
I have access to tier 3 in my API but I can't seem to access the models. Will these be supported in the near future?

**Comment by keithschacht** [comment]
@REDDKOMET I suspect that they already work and you just need to configure one within the settings of the app. I haven’t tried myself yet, but give it a try and let me know if you run into any issues. I read that o1 does not support streaming so it may seem like it’s slow, but I don’t think they changed the API for it.

**Reply by REDDKOMET** [reply]
The issue is I can't select it within the settings of the app and I tried manually adding without any luck. I can access it within the OpenAI API dashboard.

**Reply by keithschacht** [reply]
With HostedGPT, it comes pre-configured with OpenAI API already set up. I think you just need to ad this new language model. Go to settings (click your profile pic in bottom left), then click to Language Models, then click Add New. It's important that the API name be "o1-preview" since that's what the API is expecting (according to my read of the docs).

Once you have a language model you still won't see an assistant you can talk to, so now go to Assistants and click to create a new assistant. Call this assistant whatever you want but make sure it's using the o1-preview language model you just created.

Now you can chat with that assistant. Note: you must have access to the o1-preview via the OpenAI api. Most people don't, you have to be a certain tier. Also, it won't be a great experience within the app yet because OpenAI has not added streaming.

**Reply by keithschacht** [reply]
Actually, after writing all this, I tried it myself and it's not working. First I got an API error that I didn't have access, but after I switched to my proper API which has access to the model, I stopped getting an error message but the app just appears to hang without ever giving a response.

I dug in a bit and there are two issues. The o1-preview model does not yet support custom instructions, however even after I cleared those within my newly created assistant it still didn't work. I tried debugging this with curl and I think the issue is that even though the o1-preview model does not support streaming, HostedGPT is still making an API call with `streaming` set to true.

A modification to the app's code is needed. Do you know ruby to tee up a PR? I can tell you exactly what I think needs to change to make this work so you could help fix it for everyone, but I'm not sure when I can tee up this PR myself.
